sdk/attestation/azure-security-attestation/src/attestation_administration_client.cpp

```diff
@@ -58,7 +58,11 @@ AttestationAdministrationClient::AttestationAdministrationClient(
   std::vector<std::unique_ptr<HttpPolicy>> perCallpolicies;
 
   m_pipeline = std::make_shared<Azure::Core::Http::_internal::HttpPipeline>(
-      options, std::move(perRetrypolicies), std::move(perCallpolicies));
+      options,
```

Comment:
As a part of this check, I looked up the requirements for the user-agent. From [User Agent Guidelines](https://azure.github.io/azure-sdk/general_azurecore.html#user-agent-format)

```
<package_name>: client library package name as it appears to the developer, replacing slashes with dashes and removing the Azure indicator. For example, “Security.KeyVault” (.NET), “security.keyvault” (Java), “keyvault” (JavaScript & Python)
```

We had a discussion recently where we decided that "package name" for C++ is "the package name as found on vcpkg", which (for better or worse) includes the trailing `-cpp`.

I'm not a huge fan of the redundancy, but if we believe the package name for C++ packages is the package name as listed in vcpkg. For example, here's the name for the Blob service as reported by `vcpkg search`:
```
azure-storage-blobs-cpp  12.6.2           Microsoft Azure Storage Blobs SDK for C++
```

Reply:
We probably shouldn't.
Do we not have `azsdk-cpp-` user agent prefix now? It has to, per user agent guidelines: `azsdk-<sdk_language>-...`
So, `azsdk-cpp-azure-storage-blobs-cpp` looks really redundant and not hierarchical, unlike `azsdk-cpp-storage-blobs`, which is also 10 characters shorter but has the same info. We have been using that format since GA 1.0, and so the telemetry collected also uses that schema.

Reply:
If the rule for the C++ SDK is "package names match the name in vcpkg", then `-cpp` is a part of our package name.

At the last time we discussed this, we decided that the rule was "package names match the name in vcpkg" because that's what our customers experience.

I can start a discussion with the architecture board about updating the guidelines for C++ if you'd like, I honestly don't know what they'll say.

Reply:
Vcpkg package name is not much of a value, I think we should be using what makes sense in the scope of SDK instead - i.e. if in the future add it to Nuget ot winget (if they add feature for C++ libs), the naming convention there is `Microsoft.Azure...`, given that - should we be sending different telemetry strings depending on the name that was used to acquire the package?

Reply:
The guidance on user-agent is very clear: The name in user-agent is the package name. That implies that if we make a change to alter our package names, then the user-agent string is going to change.

If we end up distributing the SDK components via multiple package managers, we'll have to figure out how to deal with that when it happens. My personal take is that we should only ever be distributed via a single package management solution.

Reply:
Ok, just went to the arch board folks, the answer I got back was a bit of a hybrid. FIrst off, as you implied: Changing the name of a package is considered a breaking change. That means that we cannot change the names for the user-agent string on existing packages.

Ted on the arch board wasn't bothered by the presence of "cpp" twice in the user-agent string, especially given that the package name to vcpkg has the suffix. 

There was a general consensus that changing package managers is a "big deal" and should probably be avoided if at all possible. And similarly, if we need to live in a world with multiple package managers, we're going to have to do somewhat different surgery on the UA string (likely by adding another segment in the UA string).

The rule I'm operating under for now is "For new services, the package name used to form the user-agent (and distributed tracing library name) MUST be the package name as it appears in vcpkg; Existing services are grandfathered in to whatever name they currently are using to avoid breaking changes".

I'll be pushing an update to the PR in a few minutes that reflects this change (and modifies the tracer name to reflect the [current guidance on OpenTelemetry integration](https://github.com/Azure/azure-sdk/pull/5398)).

Reply:
Thank you Larry, that's all eventually good, but thank you for clarifying and thinking this through!
I am not trying to start the third round of discussions - I am grateful that you went and discussed it for the second time and clarified it in all details, but I'll just say, if we are going to omit `azure-` in the user-agent package name (I think I've heard about it at some point?), then the same logic could apply to `-cpp`.
In the end, honestly, it is all fine. The argument of not changing it for existing package for telemetry research purposes is actually the only really practical one, the rest is aesthetics, and there is some aesthetics in both naming schemas, as each of them is true to a rationale behind it.

Reply:
The azure core guidelines for User-Agent require the removal of "azure-" at the start of the package string, we are following the guidelines. If we strip the -cpp, then we need to update the azure core guidelines for User-Agent to allow stripping the -cpp from the package name. 

When I asked the core arch team about it, they didn't think the redundancy was an issue worth addressing.

In the future, new packages should use foo-cpp, existing packages must not change their name (so attestation is going to be "security.attestation" for the forseeable future, even though the package name is "security-attestation" (even without the -cpp).